src/target/riscv/riscv.c

```diff
@@ -1561,6 +1561,52 @@ int riscv_remove_watchpoint(struct target *target,
 	return ERROR_OK;
 }
 
+typedef enum {
+	M6_HIT_ERROR,
+	M6_HIT_NOT_SUPPORTED,
+	M6_NOT_HIT,
+	M6_HIT_BEFORE,
+	M6_HIT_AFTER,
+	M6_HIT_IMM_AFTER
+} mctrl6hitstatus;
+
+static mctrl6hitstatus check_mcontrol6_hit_status(struct target *target,
+		riscv_reg_t tdata1, uint64_t hit_mask)
+{
+	uint32_t hit0 = get_field(tdata1, CSR_MCONTROL6_HIT0);
+	uint32_t hit1 = get_field(tdata1, CSR_MCONTROL6_HIT1);
+	uint32_t hit_info = (hit1 << 1) | hit0;
```

Comment:
Pedantic detail:

The variables can be const. It will helps the code readers.

```suggestion
	const uint32_t hit0 = get_field(tdata1, CSR_MCONTROL6_HIT0);
	const uint32_t hit1 = get_field(tdata1, CSR_MCONTROL6_HIT1);
	const uint32_t hit_info = (hit1 << 1) | hit0;
```

Reply:
addressed